### PR TITLE
Kostia/feture/bread crumbs navigation

### PR DESCRIPTION
**Pull request by StoliarskyiKostiantyn**

### CODE REVIEW

File: app/controllers/breadcrumbs.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+from flask import url_for
+from flask_login import current_user
+from app import models as m, db
```
Comment (SvyatoslavArtymovych) [inline app/controllers/breadcrumbs.py]: add blank line after ` from flask_login import current_user`

File: app/controllers/breadcrumbs.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                type=s.BreadCrumbType.Collection,
+                url=url_for(
+                    "book.sub_collection_view",
+                    book_id=book_id,
+                    collection_id=collection_id,
+                ),
+                label=collection.label,
+            )
+        ]
+    if section_id != 0 and collection_path:
```
Comment (SvyatoslavArtymovych) [inline app/controllers/breadcrumbs.py]: if section_id and collection_path:

File: app/controllers/breadcrumbs.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                    url=url_for(
+                        "book.section_view",
+                        book_id=book_id,
+                        collection_id=collection_path[0],
+                        sub_collection_id=collection_path[0],
+                    ),
+                    label=section.label,
+                )
+            ]
+        if interpretation_id != 0:
```
Comment (SvyatoslavArtymovych) [inline app/controllers/breadcrumbs.py]: if interpretation_id

File: app/templates/book/collection_view.html [hunk 4]
```diff
@@ -19,12 +19,27 @@
...
   <!-- prettier-ignore -->
-  <div class="flex p-2">
-    <div>
-      <h1 class="text-l font-extrabold dark:text-white ml-4"> {{ book.owner.username }} </h1>
-      <h1 class="text-2xl font-extrabold dark:text-white ml-4">{{ book.label }}</h1>
-    </div>
-  </div>
+  <nav class="flex mt-5 mb-3" aria-label="Breadcrumb">
+  <ol class="inline-flex items-center space-x-1 md:space-x-3 ml-5">
+    {% for breadcrumb in breadcrumbs %}
```
Comment (SvyatoslavArtymovych) [inline app/templates/book/collection_view.html]: Move Breadcrumbs to separate html file

File: app/schema/breadcrumbs.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    type: BreadCrumbType
+
+    # How breadcrumbs must look like
+    # Book List    > Book Name  > Top Level Collection > SubCollection > Section        > Interpretation
+
+    # if im not owner of a book
+    # John's books > Book Name  > Top Level Collection > SubCollection > Section        > Interpretation
+
+    # if i owner
+    # My Books     > Book Title > Part I               > Chapter X     > Paragraph 1.7  > By John
```
Comment (SvyatoslavArtymovych) [inline app/schema/breadcrumbs.py]: Is it necessary?

File: app/views/book.py [hunk 6]
```diff
@@ -10,6 +10,7 @@
 
 from app.controllers import create_pagination
```
Comment (SvyatoslavArtymovych) [inline app/views/book.py]: from app.controllers import create_pagination, create_breadcrumbs

File: app/views/book.py [hunk 7]
```diff
@@ -10,6 +10,7 @@
...
 from app import models as m, db, forms as f
+from app.controllers.breadcrumbs import create_breadcrumbs
```
Comment (SvyatoslavArtymovych) [inline app/views/book.py]: remove this line

**Review by SvyatoslavArtymovych — APPROVED** [review]

**Review by kipod — APPROVED** [review]
Well done.

**Pull request merged**